Explorer/Assets/DCL/WebRequests/GenericDownloadHandlerUtils.cs

```diff
@@ -267,38 +275,60 @@ public OverwriteFromJsonAsyncOp(T target, WRJsonParser jsonParser, WRThreadFlags
 
             public async UniTask<T?> ExecuteAsync(TRequest request, CancellationToken ct)
             {
-                UnityWebRequest webRequest = request.UnityWebRequest;
-
-                string text = webRequest.downloadHandler.text;
-
-                await SwitchToThreadAsync(threadFlags);
+                DownloadHandler downloadHandler = request.UnityWebRequest.downloadHandler;
+                string text = null;
 
                 try
                 {
-                    switch (jsonParser)
+                    if (jsonParser == WRJsonParser.Unity
+#if !UNITY_EDITOR
+                        || jsonParser == WRJsonParser.NewtonsoftInEditor
+#endif
+                        )
                     {
-                        case WRJsonParser.Unity:
-                            JsonUtility.FromJsonOverwrite(text, Target);
-                            return Target;
-                        case WRJsonParser.Newtonsoft:
-                            JsonConvert.PopulateObject(text, Target!);
-                            return Target;
-                        case WRJsonParser.NewtonsoftInEditor:
-                            if (Application.isEditor)
-                                goto case WRJsonParser.Newtonsoft;
-
-                            goto case WRJsonParser.Unity;
-                        default: throw new ArgumentOutOfRangeException(nameof(jsonParser), jsonParser, null);
+                        text = downloadHandler.text;
+
+                        if ((threadFlags & WRThreadFlags.SwitchToThreadPool) != 0)
+                            await UniTask.SwitchToThreadPool();
+
+                        JsonUtility.FromJsonOverwrite(text, Target);
+                    }
+                    else
+                    {
+                        var nativeData = downloadHandler.nativeData;
+
+                        if ((threadFlags & WRThreadFlags.SwitchToThreadPool) != 0)
+                            await UniTask.SwitchToThreadPool();
+
+                        var serializer = JsonSerializer.CreateDefault();
+
+                        unsafe
+                        {
+                            var dataPtr = (byte*)nativeData.GetUnsafeReadOnlyPtr();
+
+                            using var stream = new UnmanagedMemoryStream(dataPtr, nativeData.Length,
+                                nativeData.Length, FileAccess.Read);
+
+                            using var textReader = new StreamReader(stream, Encoding.UTF8);
+                            using var jsonReader = new JsonTextReader(textReader);
+                            serializer.Populate(jsonReader, Target);
+                        }
```

Comment:
I this it to be vary similar to the function in `public struct CreateFromJsonOp<T, TRequest> : IWebRequestOp<TRequest, T> where TRequest: struct, ITypedWebRequest, IGenericDownloadHandlerRequest`. Is there a way to remove duplication?

Reply:
The only way I can think of would use lambdas, but we're trying to get rid of allocations here, not make more. I could try getting rid of the overwrite variant entirely. It does not add any value afaict.

Reply:
Actually, getting rid of the create variant might be easier.

Reply:
I'll do it later.